src/Ftp/Directory.php

```diff
@@ -11,21 +11,28 @@ class Directory
 
     private $pathParts;
 
-    public function __construct($connection, $filePath)
+    private $useFtpSiteCommand;
+
+    public function __construct($connection, $filePath, $useFtpSiteCommand)
     {
         $this->connection       = $connection;
         $this->directoryPath    = dirname($filePath);
         $this->pathParts        = array_filter(explode('/', $this->directoryPath)); // foo/bar/bat
+        $this->useFtpSiteCommand = $useFtpSiteCommand;
     }
 
     public function create()
     {
-        $pathPart = '';
-        if(!$this->exists($this->directoryPath)) {
-            foreach($this->pathParts as $part){
-                $pathPart .= '/' . $part;
-                if(!$this->exists($pathPart)){
-                    ftp_mkdir($this->connection, $pathPart);
+        if($this->useFtpSiteCommand) {
+            ftp_site($this->connection, "SITE MKDIR $this->directoryPath");
```

Comment:
use ftp_site($this->connection, 'MKDIR $this->directoryPath), you have duplicate SITE command